packages/eas-cli/src/credentials/android/actions/BuildCredentialsUtils.ts

```diff
@@ -1,14 +1,97 @@
 import assert from 'assert';
 import { nanoid } from 'nanoid';
+import ora from 'ora';
 
 import { AndroidAppBuildCredentialsFragment } from '../../../graphql/generated';
+import Log from '../../../log';
 import { getApplicationId } from '../../../project/android/applicationId';
 import { getProjectAccountName, getProjectConfigDescription } from '../../../project/projectUtils';
-import { promptAsync } from '../../../prompts';
+import { confirmAsync, promptAsync } from '../../../prompts';
 import { findAccountByName } from '../../../user/Account';
 import { Context } from '../../context';
 import { AppLookupParams } from '../api/GraphqlClient';
 
+/**
+ * Legacy credentials can be copied over to EAS if the user does not have
+ * EAS credentials set up yet
+ */
+export async function canCopyLegacyCredentialsAsync(
+  ctx: Context,
+  app: AppLookupParams
+): Promise<boolean> {
+  const appCredentials = await ctx.newAndroid.getAndroidAppCredentialsWithCommonFieldsAsync(app);
+  if (appCredentials) {
+    return false; // modern credentials already exist
+  }
+
+  const legacyAppCredentials = await ctx.newAndroid.getLegacyAndroidAppCredentialsWithCommonFieldsAsync(
+    app
+  );
+  return !!legacyAppCredentials; // user has some legacy credentials
+}
+
+export async function promptUserAndCopyLegacyCredentialsAsync(
+  ctx: Context,
+  app: AppLookupParams
+): Promise<void> {
+  assert(
+    !ctx.nonInteractive,
+    'Copying over Expo Classic credentials cannot be run in non-interactive mode'
+  );
+  const shouldCopy = await confirmAsync({
+    message: `We've detected credentials from Expo Classic (expo-cli). Would you like to copy them over to Expo Application Services (EAS)?`,
+  });
+  if (!shouldCopy) {
+    return;
+  }
+  assert(
+    await canCopyLegacyCredentialsAsync(ctx, app),
+    'User not eligible to copy Expo Classic credentials to EAS'
+  );
```

Comment:
```suggestion
  );
  assert(
    await canCopyLegacyCredentialsAsync(ctx, app),
    'User not eligible to copy Expo Classic credentials to EAS'
  );
  const shouldCopy = await confirmAsync({
    message: `We've detected credentials from Expo Classic (expo-cli). Would you like to copy them over to Expo Application Services (EAS)?`,
  });
  if (!shouldCopy) {
    return;
  }
```
I know it probably doesn't matter but I'd check if there's anything to copy before asking the user if they want to copy the credentials.